File: backend/replace.py
import os
import re

frontend_dir = 'c:\\Users\\Microsoft\\Desktop\\flotte_telephonique\\frontend\\src'

def process_file(filepath):
    if not (filepath.endswith('.tsx') or filepath.endswith('.ts') or filepath.endswith('.md')):
        return
        
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    original = content
    
    # +€950 -> +950 MAD
    content = re.sub(r'\+€([\d\s]+)', r'+\1 MAD', content)
    # €485 -> 485 MAD
    content = re.sub(r'€([\d\s]+)', r'\1 MAD', content)
    
    # 55€ -> 55 MAD
    content = re.sub(r'([\d\s]+)€', r'\1 MAD', content)
    
    # Remplacer les € restants
    content = content.replace('€', 'MAD')
    
    # Remplacer les numerots +33
    content = content.replace('+33 ', '+212 ')

    if content != original:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Updated {filepath}")

for root, dirs, files in os.walk(frontend_dir):
    for file in files:
        process_file(os.path.join(root, file))

File: frontend/replace.js
const fs = require('fs');
const path = require('path');

function walkDir(dir, callback) {
  fs.readdirSync(dir).forEach(f => {
    let dirPath = path.join(dir, f);
    let isDirectory = fs.statSync(dirPath).isDirectory();
    isDirectory ? 
      walkDir(dirPath, callback) : callback(path.join(dir, f));
  });
}

const frontendDir = 'c:\\Users\\Microsoft\\Desktop\\flotte_telephonique\\frontend\\src';

walkDir(frontendDir, function(filePath) {
  if (filePath.endsWith('.tsx') || filePath.endsWith('.ts')) {
    let content = fs.readFileSync(filePath, 'utf8');
    let original = content;

    // Remplacer "€" placé AVANT un nombre : "€485" -> "485 MAD", "+€950" -> "+950 MAD"
    content = content.replace(/\+€([\d\s]+)/g, '+$1 MAD');
    content = content.replace(/€([\d\s]+)/g, '$1 MAD');
    
    // Remplacer "€" placé APRES un nombre ou avec "/mois" : "45€" -> "45 MAD", "€55/mois" -> "55 MAD/mois"
    content = content.replace(/([\d\s]+)€/g, '$1 MAD');
    content = content.replace(/€([\d\s]+)\/mois/g, '$1 MAD/mois');
    
    // Remplacer les "€" restants en gardant un espace : "(€)" -> "(MAD)"
    content = content.replace(/€/g, 'MAD');
    
    // Remplacer "+33 6" par "+212 6"
    content = content.replace(/\+33\s/g, '+212 ');

    if (content !== original) {
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`Updated ${filePath}`);
    }
  }
});
